zarr/util.py

```diff
@@ -33,10 +33,23 @@ def flatten(arg: Iterable) -> Iterable:
 }
 
 
+class NumpyEncoder(json.JSONEncoder):
+
+    def default(self, o):
+        # See json.JSONEncoder.default docstring for explanation
+        if isinstance(o, np.integer):
+            return int(o)
+        if isinstance(o, np.floating):
+            return float(o)
+        if isinstance(o, np.ndarray):
+            return o.tolist()
```

Comment:
Interesting. Though this won't necessarily decode back to an array though. Any thoughts on that?

Reply:
I removed it, to make the json encoder more generic and not numpy specific.